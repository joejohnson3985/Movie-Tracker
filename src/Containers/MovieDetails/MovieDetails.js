import React from "react";
import { Link } from "react-router-dom";
import Back from '../../Images/back.svg';
import './MovieDetails.scss'



const MovieDetails = ({
  vote_average,
  title,
  release_date,
  poster_path,
  backdrop_path,
  overview,
  favoriteMovie,
  favorited,
  deleteFavorite,
  showPopup,
  isLoggedIn,
  id
}) => {
  console.log({ title });
  return (
    <div className="movie-details">
      <div className="movie-grid">
        <div>
          <img
            src={`https://image.tmdb.org/t/p/w500/${backdrop_path}`}
            alt="img poster"
            className="alt-poster"
          />
        </div>
        <div className="detail-text">
          <h1>{title}</h1>
          <p>{overview}</p>
        </div>
      </div>
      <div>
        <Link to="/" className="back-btn">
          <img src={Back} className="back-button" />
        </Link>
        </div>
    </div>
  );
};

export default MovieDetails;
