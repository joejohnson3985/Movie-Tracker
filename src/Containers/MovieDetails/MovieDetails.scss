.movie-details{
  display: grid;
  grid-template-columns: 1200px 1fr;
  background-color: gainsboro;
  width: 100%;
  height: 100%;

}

.movie-grid{
  display: inline;
  background-color: white;
  width: 1050px;
  display: grid;
  grid-template-columns: 1fr 400px 60px;
  margin: 50px;
  text-align: justify-all;
}

.back-button{
  margin-top: 200px;
}

.movie-buttons{
  margin: 20px;
  height: 40px
}


.alt-poster{
  margin:20px;
  margin-top: 40px;
  width: 90%;
  height: 65%
}

.detail-text{
  margin-top: 30px;
}