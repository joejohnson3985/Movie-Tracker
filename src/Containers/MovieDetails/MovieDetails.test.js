import React from 'react';
import { shallow } from 'enzyme'
import MovieDetails from './MovieDetails'



let mockSelectedMovie = {id: 1234,
  posterImage: "https://image.tmdb.org/t/p/w500/12345678.jpg",
  rating: 9.5,
  releaseDate: "2005-07-15",
  synopsis: "The quick brown fox jumps over the...",
  title: "A title"}


describe('MovieDetails', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <MovieDetails />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot when there is a selected movie', () => {
    const wrapper = shallow(
      <MovieDetails selectedMovie={mockSelectedMovie} />
    )

    expect(wrapper).toMatchSnapshot()
  })
})