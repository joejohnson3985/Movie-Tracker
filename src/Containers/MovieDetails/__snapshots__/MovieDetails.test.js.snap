// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should match snapshot 1`] = `ShallowWrapper {}`;

exports[`MovieDetails should match snapshot when there is a selected movie 1`] = `ShallowWrapper {}`;
